### PR TITLE
feat: add tracking events for filterPermissionlessCredits and filterCreditClass

### PR DESCRIPTION
**Pull request by blushi**
## Description

Closes: regen-network/regen-registry#1765

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### How to test

Go to https://deploy-preview-2065--regen-marketplace.netlify.app/projects/1 and apply some filters.
Check that the events are appearing on https://analytics.google.com/analytics/web/?authuser=1#/p340668480/realtime/overview?params=_u..nav%3Dmaui%26_u.dateOption%3Dtoday%26_u.comparisonOption%3Ddisabled&collectionId=life-cycle

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-marketplace/src/pages/Projects/Projects.SideFilter.tsx [hunk 1]
```diff
@@ -115,12 +118,18 @@ export const ProjectsSideFilter = ({
...
                                 ...creditClassSelectedFilters,
                                 [path]: event.target.checked,
-                              })
-                            }
+                              });
+                              track<
+                                'filterCreditClass',
+                                FilterCreditClassEvent
+                              >('filterCreditClass', {
+                                creditClassId: path,
```
Comment (blushi) [inline web-marketplace/src/pages/Projects/Projects.SideFilter.tsx:130]: asked @erikalogie whether we should also specify whether it's about selecting or unselecting some credit class

Comment (blushi) [reply web-marketplace/src/pages/Projects/Projects.SideFilter.tsx:131]: updated based on latest requirements

**Comment by blushi** [comment]
@erikalogie see testing instructions above

**Comment by erikalogie** [comment]
@blushi I don't see the events changing in GA

**Comment by blushi** [comment]
> @blushi I don't see the events changing in GA

Are you looking at the "Regen Marketplace [DEVELOPMENT]" app (like provided in the link in the PR description)? I can see the events populating successfully there:
image


Also make sure you don't have any privacy/ad blocker enabled.

**Comment by erikalogie** [comment]
> https://deploy-preview-2065--regen-marketplace.netlify.app/projects/1

Ah yes it was my ad blocker. Looks great!

**Review by ryanchristo — APPROVED** [review]
utACK

**Pull request merged**